code/processes/wdb.q

```diff
@@ -240,21 +240,40 @@ setcompression:{[compression] if[3=count compression;
 				]}
 resetcompression:{setcompression 16 0 0 }
 
-endofdaysortdate:{[dir;pt;tablist;hdbsettings]
+//check if the hdb directory contains current partition
+//if yes check if patition is empty and if it is not see if any of the tables exist in both the 
+//temporary parition and the hdb partition. If there is a clash abort operation otherwise copy 
+//each table to the hdb partition
+movetohdb:{[dw;hw;pt] 
+  $[not(`$string[pt])in key hsym`$-10 _ hw;
+     .[.os.ren;(dw;hw);{.lg.e[`mvtohdb;"Failed to move data from wdb ",x," to hdb directory ",y," : ",z]}[dw;hw]];
+      not any a[dw]in(a:{key hsym`$x})[hw];
+      [{[y;x] 
+        $[not(b:`$last"/"vs x)in key hsym`$y;
+          [.[.os.ren;(x;y);{[x;y;e].lg.e[`mvtohdb;"Table ",string[x]," has failed to copy to ",y," with error: ",e]}[b;y;]];
+           .lg.o[`mvtohdb;"Table ",string[b]," has been successfully moved to ",y]];
+          .lg.e[`mvtohdb;"Table ",string[b]," was skipped because it already exists in ",y]];
+        }[hw]'[dw,/:"/",/:system"ls ",dw];
```

Comment:
This should make use of `key` instead of `ls`. Can also use `sv` too build up the path too, this will avoid needing to use `hsym` in the lambda.
```
{` sv/:x,/:key x}hsym`$dw
```